spec/index.bs

```diff
@@ -2517,6 +2561,112 @@ dictionary GPUImageBitmapCopyView {
     </div>
 </div>
 
+### <dfn method for=GPUCommandEncoder>copyBufferToTexture(source, destination, copySize)</dfn> ### {#GPUCommandEncoder-copyBufferToTexture}
+
+<div algorithm="GPUCommandEncoder.copyBufferToTexture">
+
+  **Arguments:**
+    - {{GPUBufferCopyView}} |source|
+    - {{GPUTextureCopyView}} |destination|
+    - {{GPUExtent3D}} |copySize|
+
+  **Returns:** void
+
+  Encode a command into the {{GPUCommandEncoder}} that copies data from a sub-region of a
+  {{GPUBuffer}} to a sub-region of a {{GPUTexture}} [=subresource=].
+
+  |source| and |copySize| define the region of the source buffer.
+
+  |destination| and |copySize| define the region of the destination texture [=subresource=].
+
+</div>
+
+<div algorithm class=validusage>
+
+<dfn abstract-op>copyBufferToTexture Valid Usage</dfn>
+
+Given a {{GPUCommandEncoder}} |encoder| and the arguments {{GPUBufferCopyView}} |source|,
+{{GPUTextureCopyView}} |destination|, {{GPUExtent3D}} |copySize|, let
+  - |blockWidth| be the [=texel block width=] of |destination|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[format]]}}.
+  - |blockHeight| be the [=texel block height=] of |destination|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[format]]}}.
+  - |blockSize| be the [=texel block size=] of |destination|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[format]]}}.
+  - |bytes in a complete row| be |blockSize| &times; |copySize|.[=Extent3D/width=] &div; |blockWidth|.
+  - |required bytes in copy| be calculated with the following algorithm assuming all the parameters are [=valid=]. An error
+    occurs when there is any overflow in the calculation of |required bytes in copy|:
+    ```
+    GPUSize64 bytesPerRowOfTexelBlocks = (source.bytesPerRow == 0) ? bytesInACompleteRow : source.bytesPerRow;
```

Comment:
where does `bytesInACompleteRow` come from?

Reply:
Oh I mean it is |bytes in a complete row| defined just above. Maybe I should make it clearer.